resteasy/src/test/scala/com/cerner/beadledom/resteasy/ModuleProcessorSpec.scala

```diff
@@ -0,0 +1,68 @@
+package com.cerner.beadledom.resteasy
+
+import org.jboss.resteasy.spi.{Registry, ResteasyProviderFactory}
+import org.scalatest.{FunSpec, MustMatchers}
+import org.scalatest.mockito.MockitoSugar
+import com.cerner.beadledom.guice.dynamicbindings.AnnotatedModule
+import com.google.inject.Guice
+
+class ModuleProcessorSpec extends FunSpec with MockitoSugar with MustMatchers{
+
+  val registry: Registry = mock[Registry]
+  val providerFactory: ResteasyProviderFactory = mock[ResteasyProviderFactory]
+
+  describe("ModuleProcessor#new") {
+    it("throws a NullPointerException for null registry") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(null, providerFactory)
+      }
+    }
+
+    it("throws a NullPointerException for null provider factory") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, null)
+      }
+    }
+  }
+
+  describe("ModuleProcessor#processInjector") {
+    it("throws a NullPointerException for null injector") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, providerFactory).processInjector(null)
+      }
+    }
+
+    it("excludes providers that have a binding annotation") {
+      val injector = Guice.createInjector(new AnnotatedModule(classOf[TestBindingAnnotation]) {
```

Comment:
I think this should just be an `AbstractModule`.

Reply:
https://github.com/cerner/beadledom/pull/143/commits/75d364f068be5b48f12a864aa2a8bd938fb929b8